src/components/common/UserProfileEdit/UserMainSettings.tsx

```diff
@@ -0,0 +1,214 @@
+import { useMutation, gql } from '@apollo/client';
+import * as yup from 'yup';
+import { isConfusing } from '@colony/unicode-confusables-noascii';
+
+import Snackbar, { SnackbarType } from '~shared/Snackbar';
+import CopyableAddress from '~shared/CopyableAddress';
+import UserMention from '~shared/UserMention';
+import Heading from '~shared/Heading';
+import {
+  FieldSet,
+  Form,
+  FormStatus,
+  Input,
+  InputLabel,
+  Textarea,
+} from '~shared/Fields';
+import Button from '~shared/Button';
+import ConfusableWarning from '~shared/ConfusableWarning';
+
+import { updateUser } from '~gql';
+import { User } from '~types';
+import { useAppContext } from '~hooks';
+
+import styles from './UserProfileEdit.css';
+
+const MSG = defineMessages({
+  heading: {
+    id: 'users.UserProfileEdit.UserMainSettings.heading',
+    defaultMessage: 'Profile',
+  },
+  labelWallet: {
+    id: 'users.UserProfileEdit.UserMainSettings.labelWallet',
+    defaultMessage: 'Your Wallet',
+  },
+  labelUsername: {
+    id: 'users.UserProfileEdit.UserMainSettings.labelUsername',
+    defaultMessage: 'Unique Username',
+  },
+  labelEmail: {
+    id: 'users.UserProfileEdit.UserMainSettings.labelEmail',
+    defaultMessage: 'Your email',
+  },
+  labelName: {
+    id: 'users.UserProfileEdit.UserMainSettings.labelName',
+    defaultMessage: 'Your display name',
+  },
+  labelBio: {
+    id: 'users.UserProfileEdit.UserMainSettings.labelBio',
+    defaultMessage: 'Bio',
+  },
+  labelWebsite: {
+    id: 'users.UserProfileEdit.UserMainSettings.labelWebsite',
+    defaultMessage: 'Website',
+  },
+  labelLocation: {
+    id: 'users.UserProfileEdit.UserMainSettings.labelLocation',
+    defaultMessage: 'Location',
+  },
+  snackbarSuccess: {
+    id: 'users.UserProfileEdit.UserMainSettings.snackbarSuccess',
+    defaultMessage: 'Profile settings have been updated.',
+  },
+  snackbarError: {
+    id: 'users.UserProfileEdit.UserMainSettings.snackbarError',
+    defaultMessage: 'Profile settings were not able to be updated. Try again.',
+  },
+});
+
+const displayName = 'users.UserProfileEdit.UserMainSettings';
+
+interface FormValues {
+  email?: string;
+  displayName?: string;
+  bio?: string;
+  website?: string;
+  location?: string;
+}
+
+interface Props {
+  user: User;
+}
+
+const validationSchema = yup.object({
+  email: yup.string().email().nullable(),
+  bio: yup.string().nullable(),
+  displayName: yup.string().nullable(),
+  location: yup.string().nullable(),
+  website: yup.string().url().nullable(),
+});
+
+const UserMainSettings = ({ user }: Props) => {
+  const appContext = useAppContext();
+
+  const { walletAddress, profile } = user;
+  const [showSnackbar, setShowSnackbar] = useState<boolean>(false);
+  useEffect(() => {
+    if (showSnackbar) {
+      const timeout = setTimeout(() => setShowSnackbar(true), 200000);
+      return () => {
+        clearTimeout(timeout);
+      };
+    }
+    return undefined;
+  }, [showSnackbar]);
```

Comment:
Please have a look at `UserAdvanceSettings` and my comment regarding this `useEffect`. If it's actually needed, please consider moving it into its own hook